addons/sale/models/sale_order.py

```diff
@@ -649,9 +683,9 @@ def _onchange_partner_id_warning(self):
                 }
             }
 
-    @api.onchange('pricelist_id')
+    @api.onchange('pricelist_id','currency_id')
     def _onchange_pricelist_id_show_update_prices(self):
-        if self.order_line and self.pricelist_id and self._origin.pricelist_id != self.pricelist_id:
+        if self.order_line:
```

Comment:
Was thinking of not modifying anything here, but also set `show_update_pricelist` to True in the company_id onchange :thinking: